apps/anoma_node/lib/node/transaction/intent_pool.ex

```diff
@@ -155,4 +182,39 @@ defmodule Anoma.Node.Transaction.IntentPool do
       {:ok, :not_present, state}
     end
   end
+
+  @spec handle_new_nullifiers(t(), MapSet.t(binary())) :: t()
+  defp handle_new_nullifiers(state, nlfs_set) do
+    set_of_txs = state.intents
+
+    new_intents =
+      Enum.reject(set_of_txs, fn x ->
+        Stream.map(x.actions, fn x -> x.proofs end)
+        |> Stream.map(fn x ->
```

Comment:
Stream makes no sense here since you will call Enum afterwards which will resolve your entire stream anyway

Reply:
we call stream map twice, so that is ok

however I do now see that we can use Stream.reject as well

Reply:
We Stream.map twice, so one less list gets created